waku/v2/protocol/rln/group_manager/dynamic/membership_fetcher_test.go

```diff
@@ -0,0 +1,67 @@
+package dynamic
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+	"github.com/waku-org/go-waku/waku/v2/protocol/rln/contracts"
+	"github.com/waku-org/go-waku/waku/v2/protocol/rln/group_manager"
+	"github.com/waku-org/go-waku/waku/v2/protocol/rln/web3"
+	"github.com/waku-org/go-waku/waku/v2/utils"
+	"github.com/waku-org/go-zerokit-rln/rln"
+)
+
+func TestFetchingLogic(t *testing.T) {
```

Comment:
Better to add few more tests to validate membership fetch. 
Could be done in separate PR. Following are some tests i can think of: 
- Chain reorg
- Membership deletes
- simulate some failures from mock client.